### DISCUSSION TITLE
hertzbeat-collector Refactor

### DISCUSSION THREAD

**Discussion by Calvin979** [apache/hertzbeat#2357] [General]
The coding in Hertzbeat collector is now becoming more and more complicated, such as `org.apache.hertzbeat.collector.dispatch.CommonDispatcher`, which is difficult to understand and harmful for extensibility.
Shall we create a plan to refactor it?

**Comment by tomsun28** [comment]
good idea. This pr https://github.com/apache/hertzbeat/pull/2271 involves some changes. It is recommended to start optimizing after it is merged. We can start refactoring from some small optimization points first instead of overturn.

**Reply by Calvin979** [reply]
> good idea. This pr #2271 involves some changes. It is recommended to start optimizing after it is merged. We can start refactoring from some small optimization points first instead of overturn.

Got it
